service/history/timerQueueProcessor.go

```diff
@@ -654,17 +654,32 @@ Update_History_Loop:
 				}
 			} else {
 				// See if we have next timer in list to be created.
-				if !td.TaskCreated || td.ActivityID == scheduleID {
+				isHeartBeatTask := timerTask.TimeoutType == int(workflow.TimeoutType_HEARTBEAT)
+
+				// Create next timer task if we don't have one (or)
+				// if current one is HB task and we need to create next HB task for the same.
```

Comment:
Can you add comments on why you need to create the next HB task here i.e. when RecordHeartBeat() is called, only the lastUpdatedTime is modified in the mutable state but the timer task remains